From 758d8baeedeb0b67b678d68db5c024ebd624da6c Mon Sep 17 00:00:00 2001
From: Stefan Agner <stefan.agner@toradex.com>
Date: Thu, 29 Mar 2018 14:01:37 +0200
Subject: [PATCH 2/2] ifplugd: if SIOCSIFFLAGS fails with EADDRNOTAVAIL, don't
 die

When using ifplugd on a RNDIS interface with monitor mode, using
SIOCSIFFLAGS seems to fail with EADDRNOTAVAIL. Treat it the same
as if iface disappeared again.

Signed-off-by: Stefan Agner <stefan.agner@toradex.com>
---
 networking/ifplugd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/networking/ifplugd.c b/networking/ifplugd.c
index 5f1f2b3c8..e832366fa 100644
--- a/networking/ifplugd.c
+++ b/networking/ifplugd.c
@@ -359,7 +359,7 @@ static void up_iface(void)
 		/* Let user know we mess up with interface */
 		bb_error_msg("upping interface");
 		if (network_ioctl(SIOCSIFFLAGS, &ifrequest, "setting interface flags") < 0) {
-			if (errno != ENODEV)
+			if (errno != ENODEV && errno != EADDRNOTAVAIL)
 				xfunc_die();
 			G.iface_exists = 0;
 			return;
-- 
2.16.2

